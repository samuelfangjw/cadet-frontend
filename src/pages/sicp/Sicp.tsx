import * as React from 'react';
import { useEffect, useState } from 'react';
import { SicpChapter } from 'src/features/sicp/SicpTypes';

import SicpDisplay from './subcomponents/SicpDisplay';

export type SicpProps = DispatchProps & StateProps & OwnProps;
export type DispatchProps = {
  handleChangeChapter: (chapter: SicpChapter) => void;
};
export type StateProps = {
  chapter: SicpChapter;
};
export type OwnProps = {};

const Sicp: React.FC<SicpProps> = props => {
  const { chapter } = props;

  const [data, setData] = useState([]);

  const getData = () => {
    console.log("effect called");
    fetch('/sicp/test.json', {
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json'
      }
    })
      .then(response => {
          console.log(response);
          return response.json();
        })
      .then(myJson => {
          console.log(myJson);
          setData(myJson);
        });
  };

  useEffect(() => getData(), []);

  //TODO check for null json files, get correct type
  let text = "";
  data.forEach(x => text = text + x["text"]);
  const content = text + "test" + chapter;

  return (
    <SicpDisplay content={content} />
  );
};

export default Sicp;
