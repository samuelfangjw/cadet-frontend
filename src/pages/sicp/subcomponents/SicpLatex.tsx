import { MathJax } from 'better-react-mathjax';
import * as React from 'react';

type SicpLatexProps = {
  math: string;
  inline: boolean;
};

const SicpLatex: React.FC<SicpLatexProps> = props => {
  const { math, inline } = props;

  return (
    <MathJax inline={inline} dynamic={false}>
      {math}
    </MathJax>
  );
};

export default SicpLatex;
