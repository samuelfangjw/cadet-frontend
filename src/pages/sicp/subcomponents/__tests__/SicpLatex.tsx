import { MathJaxContext } from 'better-react-mathjax';
import { mount } from 'enzyme';

import { mathjaxConfig } from '../../Sicp';
import SicpLatex from '../SicpLatex';

describe('Sicp latex renders', () => {
  test('correctly block', () => {
    const props = {
      math: '1+1',
      inline: false
    };

    const tree = mount(
      <MathJaxContext version={3} config={mathjaxConfig}>
        <SicpLatex {...props} />
      </MathJaxContext>
    );
    expect(tree.debug()).toMatchSnapshot();
  });

  test('correctly inline', () => {
    const props = {
      math: '1+1',
      inline: true
    };

    const tree = mount(
      <MathJaxContext version={3} config={mathjaxConfig}>
        <SicpLatex {...props} />
      </MathJaxContext>
    );
    expect(tree.debug()).toMatchSnapshot();
  });
});
