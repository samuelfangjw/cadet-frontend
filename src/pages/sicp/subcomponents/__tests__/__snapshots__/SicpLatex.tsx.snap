// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sicp latex renders correctly block 1`] = `
"<MathJaxContext version={3} config={{...}}>
  <SicpLatex math=\\"1+1\\" inline={false}>
    <MathJax inline={false} dynamic={false}>
      <span id={[undefined]} style={{...}}>
        1+1
      </span>
    </MathJax>
  </SicpLatex>
</MathJaxContext>"
`;

exports[`Sicp latex renders correctly inline 1`] = `
"<MathJaxContext version={3} config={{...}}>
  <SicpLatex math=\\"1+1\\" inline={true}>
    <MathJax inline={true} dynamic={false}>
      <span id={[undefined]} style={{...}}>
        1+1
      </span>
    </MathJax>
  </SicpLatex>
</MathJaxContext>"
`;
